content/en/docs/languages/js/propagation.md

```diff
@@ -2,51 +2,257 @@
+[Instrumentation libraries](../libraries) like
+[`@opentelemetry/instrumentation-http`](https://www.npmjs.com/package/@opentelemetry/instrumentation-http)
+or
+[`@opentelemetry/instrumentation-express`](https://www.npmjs.com/package/@opentelemetry/instrumentation-http)
+propagate context across services for you.
 
-For example, if you have several Node.js services that communicate over HTTP,
-you can use the
-[`express`](https://www.npmjs.com/package/@opentelemetry/instrumentation-express)
-and [`http`](https://www.npmjs.com/package/@opentelemetry/instrumentation-http)
-instrumentation libraries to automatically propagate trace context across
-services for you.
+If you followed the [Getting Started Guide](../getting-started/nodejs) you can
+create a client application that queries the `/rolldice` endpoint.
 
-**It is highly recommend that you use instrumentation libraries to propagate
-context.** Although it is possible to propagate context manually, if your system
-uses libraries to communicate between services, use a matching instrumentation
-library to propagate context.
+{{% alert title="Note" %}}
 
-Refer to [Libraries](/docs/languages/js/libraries) to learn more about
-instrumentation libraries and how to use them.
+You can combine this example with the sample application from the Getting
+Started guide of any other language as well. Correlation works across
+applications written in different languages without any differences.
 
-## Manual W3C Trace Context Propagation
+{{% /alert %}}
 
-In some cases, it is not possible to propagate context with an instrumentation
-library. There may not be an instrumentation library that matches a library
-you're using to have services communicate with one another. Or you many have
-requirements that instrumentation libraries cannot fulfill, even if they exist.
+Start by creating a new folder called `dice-client` and install the required
+dependencies:
+
+{{< tabpane text=true >}} {{% tab TypeScript %}}
+
+```sh
+npm init -y
+npm install typescript \
+  ts-node \
+  @types/node \
+  undici \
+  @opentelemetry/instrumentation-undici \
+  @opentelemetry/sdk-node \
+
+# initialize typescript
+npx tsc --init
+```
+
+{{% /tab %}} {{% tab JavaScript %}}
+
+```sh
+npm init -y
+npm install undici \
+  @opentelemetry/instrumentation-undici \
+  @opentelemetry/sdk-node \
+```
+
+{{% /tab %}} {{< /tabpane >}}
+
+Next, create a new file called `client.ts` (or client.js) with the following
+content:
+
+{{< tabpane text=true >}} {{% tab TypeScript %}}
+
+```ts
+import { NodeSDK } from '@opentelemetry/sdk-node';
+import {
+  SimpleSpanProcessor,
+  ConsoleSpanExporter,
+} from '@opentelemetry/sdk-trace-node';
+import { UndiciInstrumentation } from '@opentelemetry/instrumentation-undici';
+import { request } from 'undici';
+
+const sdk = new NodeSDK({
+  spanProcessors: [new SimpleSpanProcessor(new ConsoleSpanExporter())],
+  instrumentations: [new UndiciInstrumentation()],
+});
+
+sdk.start();
+
```

Comment:
Thanks for clarifying that @trentm, I did that for the JS code but forgot to verify for the TS code that things work as expected.

I wonder if it is worth to have this explanation stored somewhere, so that we can point people to it. I wonder if we have a place in docs for it 🤔

Reply:
Is there a pattern for per-lang troubleshooting docs, perhaps? Just throwing out an idea.
Or an "advanced" section for discussing getting-started gotchas?